src/Compilers/Core/Portable/DiagnosticAnalyzer/ShadowCopyAnalyzerAssemblyLoader.cs

```diff
@@ -95,13 +116,24 @@ private void DeleteLeftoverDirectories()
             }
         }
 
-        protected override string GetPathToLoad(string fullPath)
+        protected override string PreparePathToLoad(string originalFullPath)
         {
             string assemblyDirectory = CreateUniqueDirectoryForAssembly();
-            string shadowCopyPath = CopyFileAndResources(fullPath, assemblyDirectory);
+            string shadowCopyPath = CopyFileAndResources(originalFullPath, assemblyDirectory);
+            _pathMap[originalFullPath] = shadowCopyPath;
             return shadowCopyPath;
         }
 
+        internal override string GetRealLoadPath(string originalFullPath)
+        {
+            if (!_pathMap.TryGetValue(originalFullPath, out var loadPath))
+            {
+                throw new InvalidOperationException();
```

Comment:
I take it there's reasons we don't want to just propagate the KeyNotFoundException that would occur when using the indexer?

Reply:
That is exposing an implementation detail, prefer to be a bit more resilient to change here.